webapp/src/js/controllers/navigation-confirm.js

```diff
@@ -13,6 +14,7 @@ angular.module('inboxControllers').controller('NavigationConfirmCtrl',
 
     $scope.submit = function() {
       $uibModalInstance.close();
+      $rootScope.$broadcast('formOnEdit',{value:false});
```

Comment:
This looks like the last reference to `formOnEdit` so it'd be nice to get rid of it too! Modals return a promise so the caller can tell when the submit has completed. This means you can modify the `$scope.navigationCancel` function in InboxCtrl so the `then` at the bottom also sets `$scope.enketoStatus.edited = false;`, eg:

```js
    // User wants to cancel current flow, or pressed back button, etc.
    $scope.navigationCancel = function() {
      if ($scope.enketoStatus.saving) {
        // wait for save to finish
        return;
      }
      if (!$scope.enketoStatus.edited) {
        // form hasn't been modified - return immediately
        if ($scope.cancelCallback) {
          $scope.cancelCallback();
        }
        return;
      }
      // otherwise data will be discarded so confirm navigation
      Modal({
        templateUrl: 'templates/modals/navigation_confirm.html',
        controller: 'NavigationConfirmCtrl',
        singleton: true,
      }).then(function() {
        $scope.enketoStatus.edited = false;
        if ($scope.cancelCallback) {
          $scope.cancelCallback();
        }
      });
    };
```

This means the `NavigationConfirmCtrl` can be reverted and the `formOnEdit` event can be removed completely.

Reply:
Thanks @garethbowen for the advice.I've committed those changes.